source/linear-algebra/source/01-LE/02.ptx

```diff
@@ -824,66 +824,76 @@ Can <m>\RREF(A)</m> be used to find <m>\RREF(B)</m>?
 </activity>
 
 <activity estimated-time='10' xml:id="LE2-rref-tech">
-    <statement>
+    <introduction>
         <p>
 Free browser-based technologies for mathematical computation
 are available online.
         </p>
-<ul>
-<li><p>
-     Go to <url href="https://sagecell.sagemath.org/"/>.
-    </p>
-</li>
-<li>
-    In the dropdown on the right, you can select a number of different languages.
-    Select "Octave" for the Matlab-compatible syntax used by this text.
-</li>
-<li>
-    <p>Type <c>rref([1,3,2;2,5,7])</c> and then press the <kbd>Evaluate</kbd> button to compute the <m>\RREF</m> of 
-      <m>\left[\begin{array}{ccc} 1 &amp; 3 &amp; 2 \\ 2 &amp; 5 &amp; 7 \end{array}\right]</m>.
-    </p>
-</li>
-<li>
-    <p>
+    </introduction>
+    <task>
+        <p>
+Go to <url href="https://sagecell.sagemath.org/"/>.
+        </p>
+    </task>
+    <task>
+        <p>
+In the dropdown on the right, you can select a number of different languages.
+Select "Octave" for the Matlab-compatible syntax used by this text.
+        </p>
+    </task>
+    <task>
+        <p>
+Type <c>rref([1,4,6;2,5,7])</c> and then press the <kbd>Evaluate</kbd> button to compute the <m>\RREF</m> of 
+<m>\left[\begin{array}{ccc} 1 &amp; 4 &amp; 6 \\ 2 &amp; 5 &amp; 7 \end{array}\right]</m>.
+        </p>
+    </task>
+    <task>
+        <p>
 Now try using whitespace to write out the matrix and compute <m>\RREF</m> instead:
-    </p>
-    <program>
-        <input>
-A = [1 3 2
-     2 5 7]
+        </p>
+        <program>
+            <input>
+A = [
+    1 4 6
+    2 5 7
+]
 
 rref(A)
-        </input>
-    </program>
-</li>
-</ul>
-    </statement>
-</activity>
-
-<activity estimated-time='2'>
-    <statement>
-    <p>
-In the HTML version of this text, code cells are often embedded for your convenience
-when RREFs need to be computed.
-    </p>
+            </input>
+        </program>
+    </task>
+    <task>
         <p>
-Try this out to compute
-<m>\RREF\left[\begin{array}{cc|c} 2 &amp; -3 &amp; 1 \\ 3 &amp; 0 &amp; 6 \end{array}\right]</m>.
+Add the line <c>format rat</c>
+(short for <q>format using rational numbers</q>)
+to the top of the cell and re-evaluate
+to see nice fractions in place of the decimal approximations.
         </p>
-    </statement>
+    </task>
+    <task>
+        <p>
+Visit the HTML version of this activity via <url href="https://TBIL.org/"/> to use
+the code cells embedded for your convenience below.
+        </p>
+    </task>
 </activity>
 
 <sage language="octave">
-<input>rref([2,-3,1;3,0,6])</input>
+    <input>rref([1,4,6;2,5,7])</input>
```

Comment:
The only issue here is if they evaluate the next example before this one, then this one will evaluate using format rat. You could put `format none` on the first, but that defeats the purpose of showing them what a normal input would be. I'm not really sure what to suggest here. Maybe we just leave it as is? Alternatively, you could add one last sage insert that has `format none` at the top.

Reply:
```suggestion
```

I think the slicker solution is to just remove it; I'm not sure it really adds much value if they already did the one liner above.